src/snappy_device_agents/devices/maas2/maas_storage.py

```diff
@@ -0,0 +1,561 @@
+# Copyright (C) 2023 Canonical
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Ubuntu MaaS 2.x storage provisioning code."""
+
+import logging
+import subprocess
+import collections
+import json
+import math
+
+
+logger = logging.getLogger()
+
+
+class MaasStorageError(Exception):
+    def __init__(self, message):
```

Comment:
is this needed?

you can just do 
```
class MaasStorageError(Exception):
    pass
```
the re-init with super feels odd

Reply:
Yes - good catch. Think this was an artifact from a some message passing (or something).
This way is much cleaner.